presto-spi/src/main/java/com/facebook/presto/spi/plan/PlanCheckerProviderFactory.java

```diff
@@ -13,9 +13,11 @@
  */
 package com.facebook.presto.spi.plan;
 
+import java.util.Map;
+
 public interface PlanCheckerProviderFactory
 {
     String getName();
 
-    PlanCheckerProvider create(SimplePlanFragmentSerde simplePlanFragmentSerde);
+    PlanCheckerProvider create(Map<String, String> properties, PlanCheckerProviderContext planCheckerProviderContext);
```

Comment:
Nit: can properties be a part of the context?

Reply:
technically, sure. but all other similar factory interfaces keep them separate, e.g. see [ConnectorFactory.create()](https://github.com/prestodb/presto/blob/master/presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorFactory.java#L26) or [FunctionNamespaceManagerFactory.create() ](https://github.com/prestodb/presto/blob/master/presto-spi/src/main/java/com/facebook/presto/spi/function/FunctionNamespaceManagerFactory.java#L24). I'd rather keep to that convention.